src/libs_3rdparty/QSpec/src/primitives/GTWidget.cpp

```diff
@@ -40,8 +41,17 @@ namespace HI {
 void GTWidget::click(GUITestOpStatus &os, QWidget *widget, Qt::MouseButton mouseButton, QPoint p) {
     GT_CHECK(widget != nullptr, "widget is NULL");
 
+#ifdef Q_OS_MAC
+    GTUtilsMac fakeClock;
+    fakeClock.startWorkaroundForMacCGEvents(16000, false);
+#endif
+
     if (p.isNull()) {
-        p = widget->rect().center();
+        QRect rect = widget->rect();
+        p = rect.center();
+        if (qobject_cast<QLineEdit*>(widget) != nullptr) {
```

Comment:
It's not clear what's going on. A comment is required

Reply:
sure
added #ifdef + comments